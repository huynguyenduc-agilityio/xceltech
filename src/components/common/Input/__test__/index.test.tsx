import { render, screen } from '@testing-library/react';
import { Input } from '..';

describe('Input Component', () => {
  it('renders input element correctly', () => {
    render(<Input placeholder="Enter text" />);

    const input = screen.getByTestId('input-element');

    expect(input).toBeInTheDocument();
  });

  it('handles different variants correctly', () => {
    const { rerender } = render(<Input variant="primary" />);

    expect(screen.getByTestId('input-element')).toHaveClass('bg-blue-light');

    rerender(<Input variant="secondary" />);

    expect(screen.getByTestId('input-element')).toHaveClass(
      'border-gray-muted',
    );
  });

  it('applies custom className correctly', () => {
    render(<Input className="custom-class" />);

    expect(screen.getByTestId('input-element')).toHaveClass('custom-class');
  });

  it('handles disabled state correctly', () => {
    render(<Input disabled />);

    const input = screen.getByTestId('input-element');

    expect(input).toBeDisabled();
  });

  it('should match snapshot', () => {
    const { container } = render(<Input placeholder="Enter text" />);

    expect(container).toMatchSnapshot();
  });
});
