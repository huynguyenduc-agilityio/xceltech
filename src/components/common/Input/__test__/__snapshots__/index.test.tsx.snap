// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Component should match snapshot 1`] = `
<div>
  <input
    autocomplete="off"
    class="font-productsans flex w-full px-5 py-4 text-lg text-black-smoky placeholder:text-gray-soft outline-none disabled:cursor-not-allowed disabled:opacity-50 bg-blue-light h-[54px] rounded-regular"
    data-testid="input-element"
    placeholder="Enter text"
  />
</div>
`;
